app/scripts/lib/rpc-method-middleware/handlers/wallet-createSession/handler.ts

```diff
@@ -0,0 +1,324 @@
+  bucketScopes,
+  validateAndNormalizeScopes,
+  Caip25Authorization,
+  ScopedProperties,
+  getInternalScopesObject,
+  getSessionScopes,
+  NormalizedScopesObject,
+  InternalScopeString,
+  getSupportedScopeObjects,
+} from '@metamask/multichain';
+import {
+  Caveat,
+  CaveatSpecificationConstraint,
+  invalidParams,
+  PermissionController,
+  PermissionSpecificationConstraint,
+  RequestedPermissions,
+  ValidPermission,
+} from '@metamask/permission-controller';
+import {
+  CaipChainId,
+  Hex,
+  isPlainObject,
+  Json,
+  JsonRpcRequest,
+  JsonRpcSuccess,
+} from '@metamask/utils';
+import { NetworkController } from '@metamask/network-controller';
+import {
+  JsonRpcEngineEndCallback,
+  JsonRpcEngineNextCallback,
+} from '@metamask/json-rpc-engine';
+import { PermissionNames } from '../../../../controllers/permissions';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+  MetaMetricsEventOptions,
+  MetaMetricsEventPayload,
+} from '../../../../../../shared/constants/metametrics';
+import { shouldEmitDappViewedEvent } from '../../../util';
+import { CaveatTypes } from '../../../../../../shared/constants/permissions';
+import { MESSAGE_TYPE } from '../../../../../../shared/constants/app';
+import { processScopedProperties, validateAndAddEip3085 } from './helpers';
+
+type AbstractPermissionController = PermissionController<
+  PermissionSpecificationConstraint,
+  CaveatSpecificationConstraint
+>;
+
+/**
+ * Handler for the `wallet_createSession` RPC method.
+ *
+ * @param req - The request object.
+ * @param res - The response object.
+ * @param _next - The next middleware function.
+ * @param end - The end function.
+ * @param hooks - The hooks object.
+ * @param hooks.listAccounts
+ * @param hooks.removeNetwork
+ * @param hooks.addNetwork
+ * @param hooks.findNetworkClientIdByChainId
+ * @param hooks.requestPermissionApprovalForOrigin
+ * @param hooks.sendMetrics
+ * @param hooks.metamaskState
+ * @param hooks.metamaskState.metaMetricsId
+ * @param hooks.metamaskState.permissionHistory
+ * @param hooks.metamaskState.accounts
+ * @param hooks.grantPermissions
+ */
+async function walletCreateSessionHandler(
+  req: JsonRpcRequest<Caip25Authorization> & { origin: string },
+  res: JsonRpcSuccess<{
+    sessionScopes: NormalizedScopesObject;
+    sessionProperties?: Record<string, Json>;
+  }>,
+  _next: JsonRpcEngineNextCallback,
+  end: JsonRpcEngineEndCallback,
+  hooks: {
+    listAccounts: () => { address: string }[];
+    removeNetwork: NetworkController['removeNetwork'];
+    addNetwork: NetworkController['addNetwork'];
+    findNetworkClientIdByChainId: NetworkController['findNetworkClientIdByChainId'];
+    requestPermissionApprovalForOrigin: (
+      requestedPermissions: RequestedPermissions,
+    ) => Promise<{ approvedAccounts: Hex[]; approvedChainIds: Hex[] }>;
+    sendMetrics: (
+      payload: MetaMetricsEventPayload,
+      options?: MetaMetricsEventOptions,
+    ) => void;
+    metamaskState: {
```

Comment:
I realize that this already exists, but I am confused as to why this was ever added as a hook. Being obviously not a hook

Reply:
🤷